### ISSUE TITLE
[Bug]: "data did not match any variant of untagged enum AnnouncableActivities" at crates/apub/src/activities/community/announce.rs:46

### ISSUE THREAD

**Issue by RocketDerp** [LemmyNet/lemmy#3570]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

On an Instance receiving federated content, this error is frequently being logged:

```
WARN lemmy_server::root_span_builder: data did not match any variant of untagged enum AnnouncableActivities
   0: lemmy_apub::activities::community::announce::receive
         at crates/apub/src/activities/community/announce.rs:46
```

### Steps to Reproduce

1. Setup 0.18.1 instance server
2. subscribe to communities
3. Look for error in log

### Technical Details

By adding a log line in the Rust code, discussed here on Lemmy: https://lemmy.ml/post/1888041 - captured that this is an example of the data that is generating the error:

`RawAnnouncableActivities { id: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("lemmit.online")), port: None, path: "/activities/announce/d8827593-e19b-46ab-992a-14498e7ca3b9", query: None, fragment: None }, actor: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("lemmit.online")), port: None, path: "/c/Superstonk", query: None, fragment: None }, other: {"to": Array [String("https://www.w3.org/ns/activitystreams#Public")], "object": Object {"id": String("https://lemmit.online/post/147060"), "actor": String("https://lemmit.online/u/bot"), "type": String("Page"), "attributedTo": String("https://lemmit.online/u/bot"), "to": Array [String("https://lemmit.online/c/Superstonk"), String("https://www.w3.org/ns/activitystreams#Public")], "name": String("Up to 40% off on select Switch games? Don’t mind if I do!"), "cc": Array [], "content": String("<h5>This is an automated archive made by the <a href=\"https://lemmit.online/post/14692\">Lemmit Bot</a>.</h5>\n<p>The original was posted on <a href=\"https://old.reddit.com/r/Superstonk/comments/14vbiai/up_to_40_off_on_select_switch_games_dont_mind_if/\">/r/Superstonk</a> by <a href=\"https://old.reddit.com/u/Rubicant112\">/u/Rubicant112</a> on 2023-07-09 22:13:01+00:00.</p>\n"), "mediaType": String("text/html"), "source": Object {"content": String("##### This is an automated archive made by the [Lemmit Bot](https://lemmit.online/post/14692).\nThe original was posted on [/r/Superstonk](https://old.reddit.com/r/Superstonk/comments/14vbiai/up_to_40_off_on_select_switch_games_dont_mind_if/) by [/u/Rubicant112](https://old.reddit.com/u/Rubicant112) on 2023-07-09 22:13:01+00:00.\n"), "mediaType": String("text/markdown")}, "attachment": Array [Object {"href": String("https://i.redd.it/m7gr8zi2k0bb1.jpg"), "type": String("Link")}], "image": Object {"type": String("Image"), "url": String("https://lemmit.online/pictrs/image/f15451d7-50e5-42a0-bfa2-2f502971b76a.jpeg")}, "commentsEnabled": Bool(true), "sensitive": Bool(false), "published": String("2023-07-09T22:45:09.448998+00:00"), "language": Object {"identifier": String("en"), "name": String("English")}, "audience": String("https://lemmit.online/c/Superstonk")}, "cc": Array [String("https://lemmit.online/c/Superstonk/followers")], "type": String("Announce")} }
`

### Version

BE: 0.18.1

### Lemmy Instance URL

multiple

**Comment by RocketDerp** [comment]
Here is another example of the raw data that hits this error path:

`RawAnnouncableActivities { id: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("derp.foo")), port: None, path: "/activities/announce/03ee72c7-5d28-4722-b78b-1030996873d9", query: None, fragment: None }, actor: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("derp.foo")), port: None, path: "/c/hackernews", query: None, fragment: None }, other: {"to": Array [String("https://www.w3.org/ns/activitystreams#Public")], "object": Object {"id": String("https://derp.foo/post/51203"), "actor": String("https://derp.foo/u/haxor"), "type": String("Page"), "attributedTo": String("https://derp.foo/u/haxor"), "to": Array [String("https://derp.foo/c/hackernews"), String("https://www.w3.org/ns/activitystreams#Public")], "name": String("The Undecidability of BB(748): Understanding Godel’s Incompleteness Theorems [pdf]"), "cc": Array [], "content": String("<p>There is a <a href=\"http://news.ycombinator.com/item?id=36658506\">discussion on Hacker News</a>, but feel free to comment here as well.</p>\n"), "mediaType": String("text/html"), "source": Object {"content": String("There is a [discussion on Hacker News](http://news.ycombinator.com/item?id=36658506), but feel free to comment here as well."), "mediaType": String("text/markdown")}, "attachment": Array [Object {"href": String("https://www.ingo-blechschmidt.eu/assets/bachelor-thesis-undecidability-bb748.pdf"), "type": String("Link")}], "commentsEnabled": Bool(true), "sensitive": Bool(false), "published": String("2023-07-09T22:45:04.466099+00:00"), "language": Object {"identifier": String("en"), "name": String("English")}, "audience": String("https://derp.foo/c/hackernews")}, "cc": Array [String("https://derp.foo/c/hackernews/followers")], "type": String("Announce")} }`

**Comment by sunaurus** [comment]
Closing this in favor of #3639 as that has a bit more info already
